File: algorithms/sort/shell/shell.py
import random

def shell_sort(arr, gap):
    arr_size = len(arr)
    divide = gap
    
    while gap > 0:
        for i in range(gap, arr_size):
            temp = arr[i]
            j = i
            while j >= gap and arr[j - gap] > temp:
                arr[j] = arr[j - gap]
                j -= gap
            arr[j] = temp
        gap //= divide

if __name__ == "__main__":
    original_list = [random.randint(0,100) for _ in range(100)]
    shell_sort(original_list, 13)
    print("정렬된 배열:", original_list)

File: sort/shell/Shell.java
package sort.shell;

import java.util.Arrays;
import java.util.Random;

public class Shell {

    public static void setShell(int[] arr, int gap) {
        int arrLength = arr.length;
        int divide = gap;

        while (gap>0) {
            for(int i=gap; i<arrLength; i++) {
                int tmp = arr[i];
                int j = i;
                while (gap<=j && arr[j-gap] > tmp) {
                    arr[j] = arr[j-gap];
                    j -= gap;
                } 
                arr[j] = tmp;
            }
            gap /=divide;
        }
    }

    public static void main(String[] args) {
        int listLength = 100;
        int[] originalList = new int[listLength];
        Random random = new Random();
        
        for (int i=0; i<originalList.length; i++) {
            int randNum = random.nextInt(100);
            originalList[i] = randNum;
        }

        setShell(originalList,2);
        System.out.println(Arrays.toString(originalList));
    }
}
